e2e/e2e-script.sh

```diff
@@ -73,18 +84,28 @@ addJsonToFile "subID" $SUBSCRIPTION_ID
 
 # TODO(ace): generate fresh bootstrap token since one on node will expire.
 # Check if TLS Bootstrapping is enabled(no client.crt in that case, retrieve the tlsbootstrap token)
-sleep 30s
-tlsbootstrap=$(az vmss run-command invoke \
+for i in $(seq 1 10); do
+    set +e
+    tlsbootstrap=$(az vmss run-command invoke \
                 -n $VMSS_NAME \
                 -g $MC_RESOURCE_GROUP_NAME \
                 --command-id RunShellScript \
                 --instance-id 0 \
-                --scripts "cat /var/lib/kubelet/bootstrap-kubeconfig" | \
+                --scripts "cat /var/lib/kubelet/bootstrap-kubeconfig" \
+                -ojson | \
                 jq -r '.value[].message' | \
                 grep "token" | \
                 cut -f2 -d ":" | tr -d '"'
-            )
-
+    )
+    retval=$?
+    set -e
+    if [ "$retval" -ne 0 ]; then
+        echo "retrying attempt $i"
+        sleep 10s
+        continue
+    fi
+    break;
+done
```

Comment:
In the "rare" case that the 10 attempts don't result in success, I think we have to check `retval` after the loop, or else the thing just continues. Had that happen in development once.
A statement such as `[ "$retval" -eq 0 ]` would be enough to assert success.

Reply:
Good point. Fixing.